src/Platform/Microsoft.Testing.Platform/IPC/NamedPipeBase.cs

```diff
@@ -12,9 +12,11 @@ internal abstract class NamedPipeBase
     private readonly Dictionary<Type, object> _typeSerializer = [];
     private readonly Dictionary<int, object> _idSerializer = [];
 
-    public void RegisterSerializer(INamedPipeSerializer namedPipeSerializer, Type type)
+    public void RegisterSerializer<TSerializer, TInput>()
+        where TSerializer : INamedPipeSerializer, new()
     {
-        _typeSerializer.Add(type, namedPipeSerializer);
+        INamedPipeSerializer namedPipeSerializer = new TSerializer();
```

Comment:
A minor difference here is that `Activator.CreateInstance` is now used to construct `namedPipeSerializer`.